cmd/token/cmd.go

```diff
@@ -0,0 +1,173 @@
+package token
+
+import (
+	"encoding/base64"
+	"fmt"
+	"io"
+	"os"
+	"time"
+
+	"github.com/golang-jwt/jwt/v4"
+	"github.com/spf13/cobra"
+
+	"github.com/openshift/rosa/pkg/config"
+	"github.com/openshift/rosa/pkg/output"
+	"github.com/openshift/rosa/pkg/rosa"
+)
+
+var (
+	writer io.Writer = os.Stdout
+	args   struct {
+		header    bool
+		payload   bool
+		signature bool
+		refresh   bool
+		generate  bool
+	}
+)
+
+var Cmd = NewTokenCommand()
+
+func NewTokenCommand() *cobra.Command {
+	Cmd := &cobra.Command{
+		Use:   "token",
+		Short: "Generates a token",
+		Long:  "Uses the stored credentials to generate a token.",
+		Args:  cobra.NoArgs,
+		Run:   run,
+	}
+	flags := Cmd.Flags()
+	flags.BoolVar(
+		&args.payload,
+		"payload",
+		false,
+		"Print the JSON payload.",
+	)
+	flags.BoolVar(
+		&args.header,
+		"header",
+		false,
+		"Print the JSON header.",
+	)
+	flags.BoolVar(
+		&args.signature,
+		"signature",
+		false,
+		"Print the signature.",
+	)
+	flags.BoolVar(
+		&args.refresh,
+		"refresh",
+		false,
+		"Print the refresh token instead of the access token.",
+	)
+	flags.BoolVar(
+		&args.generate,
+		"generate",
+		false,
+		"Generate a new token.",
+	)
+	output.AddFlag(Cmd)
+	return Cmd
+}
+
+func run(cmd *cobra.Command, argv []string) {
+	r := rosa.NewRuntime().WithOCM()
+	defer r.Cleanup()
+	var accessToken string
+	var refreshToken string
+	var err error
+	// Check the options:
+	count := 0
+	if args.header {
+		count++
+	}
+	if args.payload {
+		count++
+	}
+	if args.signature {
+		count++
+	}
+	if args.generate {
+		count++
+	}
+
+	if count > 1 {
+		r.Reporter.Errorf("Options '--payload', '--header', '--signature', and '--generate' are mutually exclusive")
+		os.Exit(1)
+	}
+
+	if args.generate {
+		// Get new tokens:
+		accessToken, refreshToken, err = r.OCMClient.GetConnectionTokens(15 * time.Minute)
+		if err != nil {
+			r.Reporter.Errorf("Can't get new tokens: %v", err)
+			os.Exit(1)
+		}
+	} else {
+		// Get the tokens:
+		accessToken, refreshToken, err = r.OCMClient.GetConnectionTokens()
+		if err != nil {
+			r.Reporter.Errorf("Can't get token: %v", err)
+			os.Exit(1)
+		}
+	}
```

Comment:
Can we flatten this out a bit, and move to its own function and return early
```suggestion
	accessToken, refreshToken, err = getAccessTokens(r, args.generate)
	if err != nil {
		r.Reporter.Errorf("Can't get token: %v", err)
		os.Exit(1)
	}
```
The func can be something like : 
```
func getAccessTokens(r *rosa.Runtime, generate bool) (string, string, error) {
	if generate {
		return r.OCMClient.GetConnectionTokens(15 * time.Minute)
	}
	return r.OCMClient.GetConnectionTokens()
}
```

Reply:
Changed